### PR TITLE
Continnd

### PR DESCRIPTION
**Pull request by squeezyphresh**
Move ordering now available. Just have to call move_order when we are waiting for the opponent.

### CODE REVIEW

File: src/Searcher.cpp [hunk 1]
```diff
@@ -125,9 +131,10 @@ std::vector<Node> Searcher::expand(const Node& parent,
     return children;
 }
 
-void Searcher::move_order(std::vector<Node>& nodes) {
-    // Do nothing for now
-    // TODO: Better move ordering
+void Searcher::move_order() {
+    for (auto&& moves : ordered_moves) {
+        std::sort(moves.second.begin(), moves.second.end(), Searcher::ScoreSort());
```
Comment (NigoroJr) [inline src/Searcher.cpp]: There is already a `Node::operator<` and `Node::operator>` that compares the scores. Use `std::greater<Node>()` or `std::less<Node>()`. Don't you also have to consider whether the current state is a `HOME` or an `AWAY`?

File: src/Searcher.h [hunk 2]
```diff
@@ -9,12 +9,15 @@
...
 #include <algorithm>
 #include <vector>
+#include <queue>
 
 /**
  * A class that performs alpha-beta search in the game of Domineering to find
  * the best possible move for the current turn.
  */
 
+template<typename T> using branch_queue = std::priority_queue<Node, std::vector<Node>, T>;
```
Comment (NigoroJr) [inline src/Searcher.h]: I don't think you're using this. If you don't plan on using it, might as well delete this and `#include <queue>`.

File: build/config/tournament.txt [hunk 3]
```diff
@@ -5,8 +5,7 @@ NUMCLIENTS=2
 NUMGAMES=1
 
 PORT=32101
-HOST=127.0.0.1
-
+HOST=10.33.50.190
```
Comment (NigoroJr) [inline build/config/tournament.txt]: Let's keep this `127.0.0.1` rather than an arbitrary IP address.

**Comment by NigoroJr** [comment]
(Continuing discussion here since it's relevant to this PR)

Wouldn't the sort order of the scores depend on HOME (max) or AWAY (min) because if we are AWAY we want the lowest score node to be in the front, whereas if we are HOME we want the highest so that we can maximize the pruning?

**Comment by squeezyphresh** [comment]
I didn't realize that we played as Min when we were away. Easy fix, though.

File: src/Searcher.cpp [hunk 4]
```diff
@@ -125,9 +131,13 @@ std::vector<Node> Searcher::expand(const Node& parent,
...
 
-void Searcher::move_order(std::vector<Node>& nodes) {
-    // Do nothing for now
-    // TODO: Better move ordering
+void Searcher::move_order(Who team) {
+    for (auto&& moves : ordered_moves) {
+        if (team == Who::HOME)
+            std::sort(moves.second.begin(), moves.second.end(), std::greater<Node>());
+        else
+            std::sort(moves.second.begin(), moves.second.end());
```
Comment (NigoroJr) [inline src/Searcher.cpp]:
``` cpp
auto comp = team == Who::HOME ? std::greater<Node>() : std::less<Node>();
```

would probably look better, and would make me not have to point out about using braces even for single-line if statements :grimacing:

**Comment by squeezyphresh** [comment]
That gives me compile errors because the less and greater are different types. I need to replace auto with a type that they both belong to.

**Comment by NigoroJr** [comment]
Hmm... didn't know that. Okay then, can you add the braces and `std::less<Node>()` for consistency? Other than that LGTM.

[LGTM](http://lgtm.in/i/cOSt1y0Yp)

**Comment by squeezyphresh** [comment]
OK, I tried casting them and everything but that didn't work out, so I'm just pushing changes for the version with brackets and less.

**Comment by NigoroJr** [comment]
Awesome! Nice work :+1:

**Comment by NigoroJr** [comment]
Merged into dev branch. We'll merge it into master branch once we implement and test the call to `Searcher::move_order`.

**Pull request closed without merging**